edge-runtime/lib/response.ts

```diff
@@ -197,9 +203,9 @@ export const buildResponse = async ({
       logger.withFields({ rewrite_url: rewrite }).debug('Rewrite url is same as original url')
       return
     }
-    res.headers.set('x-middleware-rewrite', relativeUrl)
+    edgeResponse.headers.set('x-middleware-rewrite', relativeUrl)
     request.headers.set('x-middleware-rewrite', target)
-    return addMiddlewareHeaders(context.rewrite(target), res)
+    return addMiddlewareHeaders(context.rewrite(target), edgeResponse)
```

Comment:
I'm not sure, but possibly here we would also need to handle middleware cookies as well? next.js test fixture doesn't seem to have case for setting cookies for rewrites, so maybe we should not touch it in this PR - just flagging it (so this comment act as potential paper trail in case there was some future report about cookies not being handled for rewrites)

Reply:
Recreated the bug quite easily with the following middleware, so decided to fix it here
```
const nextRes = NextResponse.rewrite(url)
nextRes.cookies.set("nextjs-cookie", "rewrite");
return nextRes;
```